mitodl/mit-open
# Draggable list card styles

### What are the relevant tickets?

Closes https://github.com/mitodl/hq/issues/4859

### Description (What does it do?)

Applies the list card draggable / sortable designs to the User List and Learning Path item list screens.

- Provides a draggable prop and applies the draggable design to the ol-components LearningResourceListCard and LearningResourceListCardCondensed.
- Applies to the ItemsListing when sortable (User List Listing Page, Dashboard User List Details Tab).
- Removes the @page-components LearningResourceCard and LearningResourceCardTemplate from frontends/mit-open.

### Screenshots:

Condensed, desktop:
<img width="864" alt="image" src="https://github.com/user-attachments/assets/60953609-1977-4f38-942e-cc7bb213b332">

Condensed, mobile:
<img width="495" alt="image" src="https://github.com/user-attachments/assets/f3cc9139-61ca-4354-b80b-8cae50d23400">

Default, desktop:
<img width="891" alt="image" src="https://github.com/user-attachments/assets/b348853e-7cff-492b-adbd-32ffa380b144">

Default, mobile:
<img width="499" alt="image" src="https://github.com/user-attachments/assets/3e615dd5-9ef3-4716-9131-310c26b43ea3">


### How can this be tested?


While logged in:

- Navigate to these pages, click "Reorder" drag and confirm the draggable list card designs have been applied, https://www.figma.com/design/Eux3guSenAFVvNHGi1Y9Wm/MIT-Design-System?node-id=9005-102169&m=dev
  - /userlists/<id> (or click a list from /userlists to navigate to a User List Listing Page 
  - /dashboard/#my-lists and click a list to navigate to the Dashboard User List Details Tab
  - /learningpaths/<id> (or click from a Learning Path in /learningpaths to navigate to a Learning Path Listing Page)

- Drag to a new position, click "Done ordering" and confirm that the new position is preserved after refresh.


### Additional Context



The designs show the draggable list card in condensed mode for desktop (no image, User Lists version). I've added a mobile breakpoint that adjusts the drag handle and surrounding margins to work with the mobile view of the card ("condensed, mobile" above). @steven-hatch these could do with a design review and ideally added in Figma. Let me know if you have changes or if I have missed it in Figma. Same thing for the default larger list card view on the Learning Path Listings page.

Animation on the drag handle area width is challenging as we remove and redraw the list when we enter reordering mode, ie. ItemsListing toggles ItemsListingSortable and StyledPlainList. It's marked as optional, though let's prioritise separately if the design value warrants the complexity.
